servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp

```diff
@@ -1619,6 +1616,23 @@ void RendererCanvasRenderRD::set_time(double p_time) {
 void RendererCanvasRenderRD::update() {
 }
 
+static constexpr bool is_power_of_two(uint64_t p_value) {
+	return p_value && ((p_value & (p_value - 1)) == 0);
+}
+
+static constexpr uint64_t round_up_to_alignment(uint64_t p_value, uint64_t p_alignment) {
+	DEV_ASSERT(is_power_of_two(p_alignment));
+
+	if (p_alignment == 0) {
+		return p_value;
+	}
+
+	uint64_t mask = p_alignment - 1;
+	uint64_t aligned_value = (p_value + mask) & ~mask;
+
+	return aligned_value;
+}
```

Comment:
I couldn't find these functions anywhere, but I expect they exist. If not, these should probably be moved into a universal header.

Reply:
From a quick search I couldn't find a util method for this in `core`, but it's indeed something used throughout the codebase that could be refactored in a follow-up PR.

Notably, you also added the same code to `metal_utils.h` ;)

```
$ rg "\+ .*& ~" -g'!thirdparty'
core/templates/command_queue_mt.h
342:            uint32_t alloc_size = ((sizeof(T) + 8 - 1) & ~(8 - 1));

core/os/memory.cpp
79:     p2 = (void *)(((uintptr_t)p1 + sizeof(uint32_t) + p_alignment - 1) & ~((p_alignment)-1));

modules/theora/video_stream_theora.cpp
341:            w = ((ti.pic_x + ti.frame_width + 1) & ~1) - (ti.pic_x & ~1);
342:            h = ((ti.pic_y + ti.frame_height + 1) & ~1) - (ti.pic_y & ~1);

drivers/metal/metal_utils.h
78:     uint64_t aligned_value = (p_value + mask) & ~mask;

modules/etcpak/image_compress_etcpak.cpp
186:            width = width <= 2 ? width : (width + 3) & ~3;
187:            height = height <= 2 ? height : (height + 3) & ~3;
215:            dest_mip_w = (dest_mip_w + 3) & ~3;
216:            dest_mip_h = (dest_mip_h + 3) & ~3;

modules/bmp/image_loader_bmp.cpp
78:             const uint32_t line_width = (width_bytes + 3) & ~3; // Padded to 4 bytes.

modules/basis_universal/image_compress_basisu.cpp
111:            int next_width = orig_width <= 2 ? orig_width : (orig_width + 3) & ~3;
112:            int next_height = orig_height <= 2 ? orig_height : (orig_height + 3) & ~3;
```

Reply:
@akien-mga yah, that is where I pulled it from – perhaps there is a place you might recommend I add it?

Reply:
Probably in `core/math/math_funcs.h`.

Reply:
@akien-mga the latest version removes the need for this function, so problem solved for now.